starlette_discord/client.py

```diff
@@ -13,42 +15,54 @@ class DiscordOAuthSession(OAuth2Session):
     ----------
     code: :class:`str`
         Authorization code included with user request after redirect from Discord.
+    token: :class:`Union[str, int, float]`
+        A previously generated, valid, access token to use instead of the OAuth code exchange
     client_id: :class:`int`
         Your Discord application client ID.
     scope: :class:`str`
         Discord authorization scopes separated by %20.
     redirect_uri: :class:`str`
         Your Discord application redirect URI.
     """
-    def __init__(self, code, client_id, client_secret, scope, redirect_uri):
+
+    def __init__(self, client_id, client_secret, scope, redirect_uri,
+                 code: str = None,  # obtained during login callback
+                 token: Dict[str, Union[str, int, float]] = None  # example {'access_token': 'sffd8fds90ds890fs'}
```

Comment:
we should probably make `code` and `token` keyword-only arguments.

or add a `from_code()` and `from_token()` classmethod.